src/libs/Navigation/NavigationRoot.js

```diff
@@ -40,8 +41,10 @@ class NavigationRoot extends Component {
      */
     parseAndStoreRoute(state) {
         if (!state) {
+            Navigation.resetIsNavigationReady();
             return;
         }
+        Navigation.setIsNavigationReady();
```

Comment:
Does it work ? Haven't tested this, but I'd expect `setIsNavigationReady()` to be called from inside the `PublicScreens` as well and if we are transitioning to `AuthScreens` the navigation would technically not yet be ready.

Reply:
Yes it works based on testing all the transition flows. `setIsNavigationReady()` is probably called inside the `PublicScreens`, but when the AuthScreens mount there is a moment where the navigation is not ready, and so `Navigation.resetIsNavigationReady()` is called. Then the navigation state is set and `Navigation.setIsNavigationReady()` is called.

Reply:
>when the AuthScreens mount there is a moment where the navigation is not ready, and so Navigation.resetIsNavigationReady() is called. Then the navigation state is set and Navigation.setIsNavigationReady() is called

This all feels slightly opaque. I have a good understanding of why the navigation is not "ready" i.e. we need to wait to render our new navigator before we can navigate to any screens. But the reset logic is a detail that could be easily missed by someone. 

If we end up going with this solution I would recommend we:

- Develop a better understanding of what should trigger a reset of the readiness of the navigation.
- Document this in the code with some comments so nobody messes with it and it is easy to grasp.

Reply:
>Wouldn't you agree that any time we don't have a navigation state, the navigation is not ready?

No. I don't understand what would trigger an `onStateChange` event callback with no value. The docs are pretty sparse on it besides this vague warning to not depend on the navigation state returned to the callback because it's shape could change in future releases...

https://reactnavigation.org/docs/navigation-container/#onstatechange

>Ok, I could work on that. Alternatively, what if we override the "onReady" callback of the NavigationContainer so that rather than only running when the NavigationContainer mounts, it runs whenever the navigation state changes from unset to set?

I'm not too sure what is being proposed here. Can you maybe rephrase this? 

I actually like what we had before which is take the child screen rendering as an indicator that we are ready to navigate to any other screens in the navigator. That feels more intuitive than tracking an `onStateChange` callback that is called without a state.

Reply:
Ok, IMO both are equally weird if we don't have a clear explanation for why the state is `null` and what made it send that event. But I don't have a great alternate suggestion so I think at a minimum we should document (with a comment) our observation that if the callback fires with `null` we are unable to navigate until it gets called with a `NavigationState` object.